native_client/dotnet/DeepSpeechClient/DeepSpeech.cs

```diff
@@ -13,10 +15,11 @@ namespace DeepSpeechClient
     public class DeepSpeech : IDeepSpeech
     {
         private unsafe IntPtr** _modelStatePP;
-        private unsafe IntPtr** _streamingStatePP;
-
-
-
+        
+        /// <summary>
+        /// Acoustic model status.
+        /// </summary>
+        private bool _activeModel;
```

Comment:
I understand why this is here, but I don't get why you **absolutely** need this.

Reply:
My point being that `DS_CreateModel` already gets you a pointer. Also, what happens if I need to run two models in parallel ?

Reply:
This change is mostly to avoid `System.AccessViolationException` in the case accidentally calling a released model, and for the stream also to avoid using released stream. 

`System.AccessViolationException` can crash the whole program without showing a message of the error, it also can freeze the execution. 

Now you bring it to discussion, I sort of feel over-engineering from my fear.


I can't just rely on `_modelStatePP`, even after calling the native free it still holds a valid address. The option I see here is to set it to null after the release and make sure we don't damage the GC causing a leak.

Reply:
> Also, what happens if I need to run two models in parallel ?

I don't understand this, currently `DeepSpeechClient::DeepSpeech` can only hold 1 model, if you need another model will need to create another  `DeepSpeechClient::DeepSpeech` instance.

If you want to introduce multi-model I can create another wrapper for the model, just like the stream.

Reply:
> I don't understand this, currently `DeepSpeechClient::DeepSpeech` can only hold 1 model, if you need another model will need to create another `DeepSpeechClient::DeepSpeech` instance.

That's fine by me, this is how it works for others as well.



> This change is mostly to avoid `System.AccessViolationException` in the case accidentally calling a released model, and for the stream also to avoid using released stream.
> 
> `System.AccessViolationException` can crash the whole program without showing a message of the error, it also can freeze the execution.
> 
> Now you bring it to discussion, I sort of feel over-engineering from my fear.

Well, shouldn't it be expected, that if the user of the code does not properly initialize things, it's going to throw ? I don't really see the problem here. Does .Net allows to notify that one should catch an exception like some other languages does ? Maybe that should be enough.

But adding extra logic like that feels too much like over-engineering, and this is likely to blow up in the future because of forgetting something :)